src/scripts/tokenConfig.js

```diff
@@ -0,0 +1,51 @@
+import tokenListJson from '../constants/data/json/coins.json';
+import Token from './classes/Token';
+
+const tokenConfiguration = (configFlag) => {
+  let tokenList;
+
+  switch (configFlag) {
+    case 'setup':
+      tokenList = this.initialSetup();
+      break;
+    default:
+      console.log('defaul triggered');
+  }
+  return tokenList;
+};
+
+/**
+ * initialSetup() returns an array of default tokens objects.
+ */
+initialSetup = () => {
+  let setupTokenList = [];
+  tokenListJson.forEach((tokenObject) => {
+    let token = new Token(tokenObject.symbol, tokenObject.address);
+    token.index = tokenObject.id;
+    token.type = tokenObject.type;
+    token.decimals = tokenObject.decimals;
+    token.name = tokenObject.name;
+    token.ens_address = tokenObject.ens_address;
+    token.website = tokenObject.website;
+    token.logo = tokenObject.logo.src;
+    token.email = tokenObject.support.email;
+    token.type = tokenObject.type;
+    token.blog = tokenObject.social.blog;
+    token.chat = tokenObject.social.chat;
+    token.facebook = tokenObject.social.facebook;
+    token.forum = tokenObject.social.forum;
+    token.github = tokenObject.social.github;
+    token.gitter = tokenObject.social.gitter;
+    token.instagram = tokenObject.social.instagram;
+    token.linkedin = tokenObject.social.linkedin;
+    token.reddit = tokenObject.social.reddit;
+    token.slack = tokenObject.social.slack;
+    token.telegram = tokenObject.social.telegram;
+    token.twitter = tokenObject.social.twitter;
+    token.youtube = tokenObject.social.youtube;
+    setupTokenList.push(token);
+  });
+  return setupTokenList;
+};
+
+export default tokenConfiguration;
```

Comment:
what does this do?

Reply:
In the old code how things worked was we had a coins.json file in the constants/data directory. We hardcoded the first json object in the newWalletReducer so we had a default token (Ether) for when the user initialized their app. This function is initially called when the app loads for the first time, takes whatever initial token data we have in that constants file, instantiates Token objects, and then passes that array to the Wallet reducer to initialize the newly created wallet with a set of token objects. In the old code whenever we added a new token we passed data to the newWalletReducer, which then used a string template to parse data in to a copied json object template from that coins.json constants file. This was the solution to avoid having any logic in the reducer, to remove creating an index for tokens in the reducer, and it will allow us to refactor out the rest of the main bucket with a simple function that passed a token symbol and address to this file under a different function, which will then update the set of tokens any given wallet has within the app. In the new token screen where you lookup tokens by symbol, the data return sometimes does not give a token name, it might only yield a symbol and address, thats why the Token Object only has two parameters. The rest can be add with .whateverTheProperty is. When we render out all the tokens from the tokens json file, it also has a different data structure that both the initial coins.json file, token lookup return format, and this was a solution to interact with mutliple data sources in a dynamic way, with its logic abstracted to its own function.  @GregTheGreek

Reply:
passes*

Reply:
@masterrom one of the big changes is noted here if you want to look it over